lib/worker/line-numbers.js

```diff
@@ -0,0 +1,90 @@
+function parse(file) {
+	const fs = require('fs');
+	const acorn = require('acorn');
+	const walk = require('acorn-walk');
+
+	const ast = acorn.parse(fs.readFileSync(file, 'utf8'), {
+		ecmaVersion: 11,
+		locations: true
+	});
+
+	const locations = [];
+	walk.simple(ast, {
+		CallExpression(node) {
+			locations.push(node.loc);
+		}
+	});
+
+	// Walking is depth-first, but we want to sort these breadth-first.
+	locations.sort((a, b) => {
+		if (a.start.line === b.start.line) {
+			return a.start.column - b.start.column;
+		}
+
+		return a.start.line - b.start.line;
+	});
+
+	return locations;
+}
+
+function findTest(locations, declaration) {
+	// Find all calls that span the test declaration.
+	const spans = locations.filter(loc => {
+		if (loc.start.line > declaration.line || loc.end.line < declaration.line) {
+			return false;
+		}
+
+		if (loc.start.line === declaration.line && loc.start.column > declaration.column) {
+			return false;
+		}
+
+		if (loc.end.line === declaration.line && loc.end.column < declaration.column) {
+			return false;
+		}
+
+		return true;
+	});
+
+	// Locations should be sorted by source order, so the last span must be the test.
+	return spans.pop();
+}
+
+const range = (start, end) => new Array(end - start + 1).fill(start).map((element, index) => element + index);
```

Comment:
Break this guy out (since it's duplicated)? Or three strikes?

Reply:
Yea I didn't think it was worth extracting at this point.